framework/src/utils/InputParameters.C

```diff
@@ -1251,6 +1251,15 @@ InputParameters::getGroupParameters(const std::string & group) const
   return names;
 }
 
+std::set<std::string>
+InputParameters::getParametersList() const
+{
+  std::set<std::string> param_set;
+  for (auto it = _params.begin(); it != _params.end(); ++it)
+    param_set.emplace(it->first);
+  return param_set;
```

Comment:
Do you benefit from working with the `set` over a `vector`? Because you are populating this from a `map`, the elements of the vector/set are already guaranteed to be unique and sorted. With a vector you could reserve the size and then `push_back` is amortized constant whereas `set` insertion is O(log(N))

Reply:
we dont expect to work with anything where the asymptotic regime is reached here

Reply:
In terms of "I expect this API to return something unique" we are in the "it should be unique" realm, so a set is not shocking to me

Reply:
> we dont expect to work with anything where the asymptotic regime is reached here

I don't know what this means. The algorithm is always O(log(N)). It's a binary search no matter the size of the container. For parameter lists of 30 items I might expect these costs to be non-negligible if you do it a lot. But probably negligible when compared to the expensive things we do in MOOSE

Reply:
Let's change this to a vector. I think that a developer who calls `getParametersList` will just from the name assume it's unique. And if it's not obvious, then you can state it in the doxygen